hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/RaftNodeStatus.java

```diff
@@ -49,15 +49,17 @@
     STEPPED_DOWN,
 
     /**
-     * When a Raft group is being terminated, all Raft nodes' statuses in that
-     * group become {@code TERMINATING}. Once termination process is completed,
-     * then the statuses become {@link #TERMINATED}.
+     * When a Raft group is being destroyed, all Raft nodes' statuses in that
+     * group become {@code TERMINATING}. Once the Raft group destroy process
+     * is completed, then the statuses become {@link #TERMINATED}.
      */
     TERMINATING,
 
     /**
-     * When a Raft group is terminated completely, all Raft nodes' statuses in
-     * that group become {@code TERMINATED}.
+     * When a Raft group is destroyed, all Raft nodes' statuses in that group
+     * become {@code TERMINATED} after the destroy process is done. Moreover,
+     * a Raft node can move to {@code TERMINATED} status if its JVM is being
```

Comment:
It doesn't have to be JVM shutting down. This should be, "... if the `HazelcastInstance` which `RaftNode` belongs to shutting down."

Reply:
That is correct. I didn't want to refer to `HazelcastInstance` from the core Raft code Javadoc but we already have leaks in a few places so I can refer to `HazelcastInstance` here as well.

Reply:
Or you can just say Hazelcast member/node without linking the class.